### ISSUE TITLE
[Bug Report] Stash is not scanning all videos . It's Really a big issue as fundamental feature is not working properly.

### ISSUE THREAD

**Issue by gkddev** [stashapp/stash#1560]
**Describe the bug**
A clear and concise description of what the bug is:

Stash is not able to scan all the videos from a parent folder having large number of child folders. I can literally see the video on the drive but when I try to search it on stash , it's not there. Some videos from same folder exists in stash. Stash scanning is partially working.

As you see in the screenshot below, my "Original Files No Duplicate" Folder contains , 1.3TB of data alone , and this same parent folder ,I added to configuration files along with other folders in different drive. But total stash scanned files estimates 958GB only. I even tried to add a child folder from "Original Files No Duplicate" and scanned again, but I still did not found the missing files which is available in drive locally but not in stash.

**Expected behavior**
In my case , total folder which is added into configuration ,its total size exceed 1.5TB but total 958gb of files are scanned only.

**Screenshots**
![Screenshot (676)](https://user-images.githubusercontent.com/39620074/124722483-ba76d280-df27-11eb-819d-0e7905cc96b7.png)
![Screenshot (677)](https://user-images.githubusercontent.com/39620074/124722488-bba7ff80-df27-11eb-8a55-a5d291608a34.png)
![Screenshot (678)](https://user-images.githubusercontent.com/39620074/124722511-c06cb380-df27-11eb-9866-240ca0dfea2f.png)

**Stash Version: (from Settings -> About):**
0.8.0

**Desktop (please complete the following information):**
 - OS: Windows 10 PRO
 - Browser: Chrome
 - Version :Latest

**Comment by HereIam80** [comment]
Are you sure that the files you want to scan are not in fact duplicates of scenes you already have in stash? What dos the log say? If there are duplicate files, stash will recognize those as duplicates and not add them.

**Comment by gkddev** [comment]
I think while scanning the file,  stash is finding the files to be duplicate even though there is no duplicate present.

Reason:

I deleted a child folder  from the parent folder "Original Files No Duplicate" (I basically moved one of the child folder to another drive ,and made sure that the new drive is not in the configuration list) and performed "clean" task under maintenance and three files got deleted , even though child folder contains 22 items.

I again  moved the same 22 files to the child folder and scanned again "using selective scan" and now log said only three new files found. How can this happen?

But I can promise you that same file is not present anywhere in the drive. I checked manually.  

I searched same file which is present locally in the stash portal but not found. I searched using performer name, any tag, or file name. There is no file with same content available on stash but is on local drive.

**Comment by HereIam80** [comment]
My thinking is that only three of those files are actually not duplicates and added to the library during a scan. The other 19 files are duplicates of other files, which are in the DB already and thus not get cleaned or added.

But we would need to see a log to confirm.

The way stash works is that it compares the files not by filename but by hash. If stash detects an identical file with the same hash it doesn't get added - no matter the filename or location.

**Comment by gkddev** [comment]
Even if it is duplicate, one of the duplicate files should be in the stash, right? Why isn't it available then?

Anyways, So how can I get you the log for those 19 files, according to you "duplicate files"? I have 4.5k items, how to find the log for duplicate files?

**Comment by HereIam80** [comment]
What does the log say? All duplicates should be mentioned there. Look at configuration => logs after a scan.

**Comment by gkddev** [comment]
Is there a way to delete duplicate files?

**Comment by HereIam80** [comment]
If they are identical and not get added to the DB there is unfortunately no way at the moment. You have to go through your log and then delete them manually.

I think I saw a script once that could read your log and then delete the duplicates but I don't know where it is.

**Comment by gkddev** [comment]
I scanned whole drive again "Full Scan",  log did mentioned about some duplicate files but not these 19 files. There was no log about these 19 files.

**Comment by gkddev** [comment]
During full scanning , log received error :
 FFProbe encountered an error with file_name.mp4 
Error JSON:
{
    "error": {
        "code": -2,
        "string": "No such file or directory"
    }
}

Error: exit status 1

This file is available in file explorer but when I tried to rename the same file in file explorer , nothing happened, I mean no new name was reflected in the filename in file explorer. 
I moved the file to different destination and then "rename" worked and I started "clean" maintenance task then removed the same file to previous location and now I scanned again and 70 new files were found.

**Comment by gkddev** [comment]
But Still those 19 files are not showing , not even as duplicate. Why?

**Comment by gkddev** [comment]
Now I have found the problem in scanning files. Why certains files were being ignored by stash because video extensions allowed in stash configuration did not have ".ts" file for default.

only these extensions "m4v, mp4, mov, wmv, avi, mpg, mpeg, rmvb, rm, flv, asf, mkv, webm" for videos were given by default. Juts add "ts" to it and done.

description says "Comma-delimited list of file extensions that will be identified as videos." . so yeah problem sloved and now I scanned again and stash recognized new 4.5k files.
